src/server/routes/weatherLocation.js

```diff
@@ -0,0 +1,219 @@
+						required: ['latitude', 'longitude'],
+						properties: {
+							latitude: { type: 'number', minimum: '-90', maximum: '90' },
+							longitude: { type: 'number', minimum: '-180', maximum: '180' }
+						}
+					},
+					{ type: 'null' }
+				]
+			},
+			note: {
+				oneOf: [
+					{ type: 'string' },
+					{ type: 'null' }
+				]
+			}
+		}
+	}
+	const paramsValidationResult = validate(params, validParams);
+	return { valid: paramsValidationResult.valid, errors: paramsValidationResult.errors };
+}
+
+/**
+ * Fetches and inserts weather data for a newly created weather location.
+ * If the location has no existing weather data, it retrieves data starting from the earliest reading timestamp.
+ * @param location - Instance of WeatherLocation.
+ * @param conn - The database connection.
+ */
+async function addWeatherDataForLocation(location, conn) {
+	// Find last timestamp for that location 
+	let earliestDate = await WeatherData.getLatestTimeStamp(location.id, conn);
+
+	// If no weather data exists, use earliest reading timestamp
+	if (earliestDate === null) {
+		earliestDate = await Reading.getEarliestTimeStamp(conn);
+		if (earliestDate === null) {
+			log.warn('addWeatherDataForLocation(): Cannot find earliest time from WeatherData or Reading.');
+			return;
+		}
+	}
+
+	// this only gets the data for the current day(?)
+	// earliestDate = earliestDate.format('YYYY-MM-DD');
+	// const latestDate = moment().format('YYYY-MM-DD');
+	earliestDate = '2025-12-11'; // test date
+	const latestDate = moment().format('YYYY-MM-DD');
+
+	// Fetch weather data for this location's coordinates
+	const weatherData = await fetchWeatherData(
+		location.gps.latitude,
+		location.gps.longitude,
+		earliestDate,
+		latestDate
+	);
+
+	// Insert weather data records
+	if (weatherData) {
+		await conn.tx(async t => {
+			for (const data of weatherData) {
+				// TODO Verify that data.time is start of the weather value and always getting one hour of data. 
+				const endDate = data.time.clone().add(1, 'hours');
+				const newData = new WeatherData(
+					location.id,
+					data.time,
+					endDate,
+					data.temperature
+				);
+				await newData.insert(t);
+			}
+		});
+	}
+}
+
+/**
+ * Route for POST add weather location.
+ */
+router.post('/addWeatherLocation', adminAuthMiddleware('add weather locations'), async (req, res) => {
+	const response = validateWeatherLocationParams(req.body);
+	if (!response.valid) {
+		log.warn(`Got request to edit a weather location with invalid weather data, errors: ${response.errors}`);
+		failure(res, 400, 'validation failed with ' + response.errors.toString());
+	} else {
+		const conn = getConnection();
+		try {
+			const newLocation = new WeatherLocation(
+				undefined, // id
+				req.body.identifier,
+				req.body.gps ? new Point(req.body.gps.longitude, req.body.gps.latitude) : null,
+				req.body.note
+			);
+			await newLocation.insert(conn);
+			await addWeatherDataForLocation(newLocation, conn);
```

Comment:
When you add a new weather location I get a failure notice but the location is added. However, I don't think any weather data is added. I believe the issue is that the insert happens on the line above but the id is not captured and set in the newLocation so getLatestTimeStamp does not find it and it fails.